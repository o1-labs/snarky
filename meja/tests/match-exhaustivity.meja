type x('a) = {a: 'a, b: int, c: bool, d: option('a), e: [`A(int) | `B(bool)]};

let f = fun (x) => {
  switch (x) {
    | {a: {a: 15}, b, c: true, d: None, e: `B(true)} => x
    | {a: {a: 15}, b, c: true, d: Some({b: 15})} => x
  };
};
