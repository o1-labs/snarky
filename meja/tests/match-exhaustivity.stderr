File "tests/match-exhaustivity.meja", line 4, characters 2-130:
Error: This pattern-matching is not exhausive.
Here are some examples of cases that are not matched:
{d: None | Some ({b: _}), e: `A (_) | `B (false)}
{d: Some ({b: _}), e: `B (true)}
{c: false, d: None, e: `B (true)}
{a: {a: _}, c: true, d: None, e: `B (true)}
